atxm/machine.py

```diff
@@ -118,40 +153,67 @@ def _handle_errors(self, *args, **kwargs):
         self.log.warn("[recovery] restarting transaction machine!")
         self._start(now=False)
 
-    def _cycle(self) -> None:
-        """Execute one cycle"""
+    # State Transitions
+    @_transition_to_paused.before
+    def _enter_pause_mode(self):
+        self.log.info("[pause] pause mode activated")
+        return
+
+    @_PAUSED.enter
+    def _process_paused(self):
+        # TODO what action should be taken to check if we leave the pause state?
+        #  Perhaps a call to self.__strategize()
+        return
+
+    @_IDLE.enter
```

Comment:
I cannot find the docs for this usage of state instances, do you have a reference?

Reply:
See https://python-statemachine.readthedocs.io/en/latest/actions.html?highlight=.enter#actions

I used the decorator syntax - https://python-statemachine.readthedocs.io/en/latest/actions.html?highlight=.enter#bind-state-actions-using-decorator-syntax